### PR TITLE
Show approximate date since commit on commit search results

### PR DESCRIPTION
**Pull request by attfarhan**
Fixes https://github.com/sourcegraph/sourcegraph/issues/1311

### CODE REVIEW

**Comment by slimsag** [comment]
I'll review this tomorrow (remind me if I forget)

File: cmd/frontend/graphqlbackend/search_commits_test.go [hunk 1]
```diff
@@ -66,13 +68,13 @@ func TestSearchCommitsInRepo(t *testing.T) {
...
-				author: *toSignatureResolver(&git.Signature{}),
+				author: *toSignatureResolver(&gitSignatureWithDate),
 			},
 			diffPreview: &highlightedString{value: "x", highlights: []*highlightedRange{}},
 			icon:        "data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48IURPQ1RZUEUgc3ZnIFBVQkxJQyAiLS8vVzNDLy9EVEQgU1ZHIDEuMS8vRU4iICJodHRwOi8vd3d3LnczLm9yZy9HcmFwaGljcy9TVkcvMS4xL0RURC9zdmcxMS5kdGQiPjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgdmVyc2lvbj0iMS4xIiB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCI+PHBhdGggZD0iTTE3LDEyQzE3LDE0LjQyIDE1LjI4LDE2LjQ0IDEzLDE2LjlWMjFIMTFWMTYuOUM4LjcyLDE2LjQ0IDcsMTQuNDIgNywxMkM3LDkuNTggOC43Miw3LjU2IDExLDcuMVYzSDEzVjcuMUMxNS4yOCw3LjU2IDE3LDkuNTggMTcsMTJNMTIsOUEzLDMgMCAwLDAgOSwxMkEzLDMgMCAwLDAgMTIsMTVBMywzIDAgMCwwIDE1LDEyQTMsMyAwIDAsMCAxMiw5WiIgLz48L3N2Zz4=",
 			label:       "[repo](/repo) &nbsp; [](/repo/-/commit/c1) &nbsp; [](/repo/-/commit/c1)",
 			url:         "/repo/-/commit/c1",
-			detail:      "c1",
+			detail:      fmt.Sprintf("[`c1` %s](/repo/-/commit/c1)", timeago.NoMax(timeago.English).Format(gitSignatureWithDate.Date)),
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/search_commits_test.go]: For test data it is usually clearer to hardcode the string you want.

File: cmd/frontend/graphqlbackend/search_commits_test.go [hunk 2]
```diff
@@ -66,13 +67,13 @@ func TestSearchCommitsInRepo(t *testing.T) {
...
 				oid:    "c1",
-				author: *toSignatureResolver(&git.Signature{}),
+				author: *toSignatureResolver(&gitSignatureWithDate),
 			},
 			diffPreview: &highlightedString{value: "x", highlights: []*highlightedRange{}},
 			icon:        "data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48IURPQ1RZUEUgc3ZnIFBVQkxJQyAiLS8vVzNDLy9EVEQgU1ZHIDEuMS8vRU4iICJodHRwOi8vd3d3LnczLm9yZy9HcmFwaGljcy9TVkcvMS4xL0RURC9zdmcxMS5kdGQiPjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgdmVyc2lvbj0iMS4xIiB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCI+PHBhdGggZD0iTTE3LDEyQzE3LDE0LjQyIDE1LjI4LDE2LjQ0IDEzLDE2LjlWMjFIMTFWMTYuOUM4LjcyLDE2LjQ0IDcsMTQuNDIgNywxMkM3LDkuNTggOC43Miw3LjU2IDExLDcuMVYzSDEzVjcuMUMxNS4yOCw3LjU2IDE3LDkuNTggMTcsMTJNMTIsOUEzLDMgMCAwLDAgOSwxMkEzLDMgMCAwLDAgMTIsMTVBMywzIDAgMCwwIDE1LDEyQTMsMyAwIDAsMCAxMiw5WiIgLz48L3N2Zz4=",
 			label:       "[repo](/repo) &nbsp; [](/repo/-/commit/c1) &nbsp; [](/repo/-/commit/c1)",
 			url:         "/repo/-/commit/c1",
-			detail:      "c1",
+			detail:      "[`c1` about a second ago](/repo/-/commit/c1)",
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/search_commits_test.go]: aah, this likely has the possibility of being non deterministic if the test runs slowly for some reason (more common than you think). Maybe the gitSignatureWithDate should be `time.Now().Sub(time.Hour * 25)` and then check for `about a day ago`

Comment (attfarhan) [reply cmd/frontend/graphqlbackend/search_commits_test.go]: Fixed: https://github.com/sourcegraph/sourcegraph/pull/1345

**Pull request merged**